pkg/operator2/oauth.go

```diff
@@ -136,3 +148,172 @@ func (c *osinOperator) handleOAuthConfig(configOverrides []byte) (*corev1.Config
 		},
 	}, nil
 }
+
+func convertProviderConfigToOsinBytes(providerConfig *configv1.IdentityProviderConfig) []byte {
+	var providerConfigBytes bytes.Buffer
+
+	bytesWriter := bufio.NewWriter(&providerConfigBytes)
+	serializer := kubejson.NewYAMLSerializer(kubejson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	switch providerConfig.Type {
+	case configv1.IdentityProviderTypeBasicAuth:
+		basicAuthConfig := providerConfig.BasicAuth
+		p := osinv1.BasicAuthPasswordIdentityProvider{
+			RemoteConnectionInfo: configv1.RemoteConnectionInfo{
+				URL: basicAuthConfig.URL,
+				CA:  getLocalConfigMapFilename(basicAuthConfig.CA),
+				CertInfo: configv1.CertInfo{
+					CertFile: getLocalSecretFilename(basicAuthConfig.TLSClientCert),
+					KeyFile:  getLocalSecretFilename(basicAuthConfig.TLSClientKey),
+				},
+			},
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeAllowAll:
+		// TODO: is the type necessarily set or would it rather be just the bool?
+		serializer.Encode(&osinv1.AllowAllPasswordIdentityProvider{}, bytesWriter)
+
+	case configv1.IdentityProviderTypeDenyAll:
+		// TODO: is the type necessarily set or would it rather be just the bool?
+		serializer.Encode(&osinv1.DenyAllPasswordIdentityProvider{}, bytesWriter)
+
+	case configv1.IdentityProviderTypeHTPasswd:
+		p := osinv1.HTPasswdPasswordIdentityProvider{
+			File: getLocalSecretFilename(providerConfig.HTPasswd.FileData),
+		}
+		serializer.Encode(&p, bytesWriter)
+
+	case configv1.IdentityProviderTypeLDAP:
+		ldapConfig := providerConfig.LDAP
+		p := osinv1.LDAPPasswordIdentityProvider{
+			URL:    ldapConfig.URL,
+			BindDN: ldapConfig.BindDN,
+			BindPassword: configv1.StringSource{
+				StringSourceSpec: configv1.StringSourceSpec{
+					// TODO: this could be Env, Value, File, either can be encrypted
```

Comment:
We have a bunch of options for `StringSources`, do we just go with the secrets being files?

Reply:
Yes, expect all secrets to be files on disk.  The config needs to be safe to put in a config map.  Make a helper to reduce `configv1.StringSource` boilerplate.